Fixed non-working delete on finna additions

humble-ahitofel opened the pull request:
When trying to delete customdata from finna organisation, it would result on attempt to delete the finna organisation itself (By attempt, I mean it failed). Now fixed in this pull request.
Also deleting weblinks from finna organisation was not working (something to do with the Weight's). Now they work, but after deleting  a link, you are redirected to the global weblink listing? (still needs some work)

humble-ahitofel committed:
Fixed non-working delete on finna additions

ajsalminen commented on src/Doctrine/EventListener/WeightInitializer.php:
```diff
@@ -92,16 +92,24 @@ private function getRepository(LifecycleEventArgs $args)
     private function loadRelatedEntities(LifecycleEventArgs $args) : ArrayCollection
     {
-        $entities = $args->getEntityManager()->createQueryBuilder()
+        $builder = $args->getEntityManager()->createQueryBuilder()
```
Could have a variable for the entity here (below) to make this a bit cleaner?

ajsalminen commented on src/Module/Finna/Controller/FinnaController.php:
```diff
@@ -410,6 +410,48 @@ public function editCustomData(Request $request, FinnaAdditions $finna_organisat
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+
```
does the following work like this despite other cases requiring jumping more hoops for Doctrine?

ajsalminen commented on src/Module/Finna/Controller/FinnaController.php:
```diff
@@ -410,6 +410,48 @@ public function editCustomData(Request $request, FinnaAdditions $finna_organisat
+
+        if ($form->isSubmitted() && $form->isValid()) {
+
+            $data = $finna_organisation->getCustomData();
```
Maybe mirror the naming in the other methods dealing with customdata and call this entries instead of data?

ajsalminen commented on src/Doctrine/EventListener/WeightInitializer.php:
```diff
@@ -92,16 +92,24 @@ private function getRepository(LifecycleEventArgs $args)
+            $builder->where('e.parent = :library')
+                ->setParameter('library', $args->getEntity()->getParent());
+        }
+        
```
This whole thing above could maybe use a clarifying comment.

humble-ahitofel committed:
Fixed for using wrong relation column
- WeightInitializer waas using 'id' as relation column for
  FinnaOrganisationWebsiteLink entity. Now using the
  correct relation column id 'finna_organisation' .

humble-ahitofel committed:
Renamed  $data to match $entries to match
common variable naming scheme.
